R/repo-helper.R

```diff
@@ -0,0 +1,75 @@
+#' Create example repo for fledge demos
+#'
+#' @param open Whether to open the new project.
+#' @param pkg Package name.
+#' @param name Name for DESCRIPTION and git.
+#' @param email Email for DESCRIPTION and git.
+#' @param date String of time for DESCRIPTION and git.
+#' @param tempdir Directory within which to create the mock package folder.
+#' @param news If TRUE, create a NEWS.md file.
+#'
+#' @return
+#' @export
+#'
+create_fledge_mock_repo <- function(open = rlang::is_interactive(),
+                               pkg = "tea",
+                               name = "Kirill M\u00fcller",
+                               email = "mail@example.com",
+                               date = "2020-12-12",
+                               tempdir = withr::local_tempdir(
+                                 pattern = "fledge",
+                                 .local_envir = parent.frame(n = 2)
+                                 ),
+                               news = FALSE
+                              ) {
+
+  withr::local_options(usethis.quiet = TRUE)
+
+  set_usethis_desc(name = name, email = email, date = date)
+  pkg <- usethis::create_package(
+    file.path(tempdir, pkg),
+    fields = list(Date = as.Date(date)),
+    open = open
+  )
+
+  withr::local_dir(new = pkg)
+  gert::git_init()
+  gert::git_config_set("user.name", name)
+  gert::git_config_set("user.email", email)
+  gert::git_add(".")
+  gert::git_commit("First commit")
+  if ("master" %in% gert::git_branch_list()$name) {
```

Comment:
not possible to create the right one directly (can't set a local git option without a git repo, and can't set a global git option + no relevant argument for `gert::git_init()`).

Reply:
Should we add an argument for the name of the new branch, and test if the current branch name is different?

Reply:
Will we need this in a demo?

Reply:
I now see we have `git_branch_move()`, maybe we don't need an argument but something like:

```suggestion
  current_branch <- gert::git_branch_list()$name
  if (current_branch != "main") {
    gert::git_branch_move(current_branch, "main")
```

(Careful, still need to delete the next line.)